integration/helpers.go

```diff
@@ -54,32 +54,56 @@ func parseOutput[T any](output string, normalize func(*T)) ([]*T, error) {
 	return ret, nil
 }
 
-// ExpectAllToMatch verifies that the expectedEntry is matched by all the
-// entries in the output.
-func ExpectAllToMatch[T any](output string, normalize func(*T), expectedEntry *T) error {
-	entries, err := parseOutput(output, normalize)
-	if err != nil {
-		return err
+func parseArrayOutput[T any](output string, normalize func(*T)) ([]*T, error) {
```

Comment:
I suggest making the names more descriptive. Something like this, maybe with one line of description:

```golang
func parseMultiJSONOutput[T any](output string, normalize func(*T)) ([]*T, error)
```

```golang
func parseJSONArrayOutput[T any](output string, normalize func(*T)) ([]*T, error) {
```